### PR TITLE
Integration4 t6 f2 34 login/signup

### PR DESCRIPTION
**Pull request by nala-he**

### CODE REVIEW

**Review by bovornc — APPROVED** [review]
Works great!

**Review by t1yena — APPROVED** [review]
lookgs good!!

File: controllers/AuthenticationController.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import UserDao from "../daos/UserDao";
+import {Express} from "express";
+
+export default class AuthenticationController {
```
Comment (MaceeQi) [inline controllers/AuthenticationController.ts:4]: I think we need documentation for this file

File: controllers/RestaurantController.ts [hunk 2]
```diff
@@ -46,6 +46,7 @@ export default class RestaurantController implements RestaurantControllerI {
             app.delete('/api/restaurants/name/:name/delete',
                 RestaurantController.restaurantController.deleteRestaurantsByRestaurantName);
             app.get('/api/restaurants/name/:name', RestaurantController.restaurantController.findRestaurantsByName);
+            app.delete('/api/restaurants/users/:uid', RestaurantController.restaurantController.deleteRestaurantByOwner);
```
Comment (MaceeQi) [inline controllers/RestaurantController.ts:49]: do you mind adding this new http endpoint to the jsdoc?

**Review by MaceeQi — COMMENTED** [review]
Left a few comments regarding jsdoc, but otherwise, looks great! :)

File: controllers/RestaurantController.ts (hunk 2 above)
Comment (nala-he) [reply controllers/RestaurantController.ts:49]: added in the restaurant testing branch

File: controllers/AuthenticationController.ts (hunk 1 above)
Comment (nala-he) [reply controllers/AuthenticationController.ts:4]: Added into the restaurant testing branch

**Review by MaceeQi — APPROVED** [review]
Thanks for adding the documentation!

**Pull request merged**